File: app/src/main/java/net/emilla/measure/FahrenheitConversion.java
package net.emilla.measure;

public record FahrenheitConversion(double degrees, boolean wasFromKelvin) {
    public double convert() {
        double offset = wasFromKelvin
            ? -459.67
            : 32.0
        ;
        return degrees * 9.0 / 5.0 + offset;
    }
}

File: app/src/main/java/net/emilla/lang/measure/FahrenheitConversion.kt
package net.emilla.lang.measure

data class FahrenheitConversion(@JvmField val degrees: Double, @JvmField val fromKelvin: Boolean) {

    fun convert(): Double = if (fromKelvin) degrees - 459.67 else degrees * 9.0 / 5.0 + 32.0
}
